src/graceful_termination.jl

```diff
@@ -14,8 +14,7 @@ _deputy_ipc_dir() = get(tempdir, ENV, "DEPUTY_IPC_DIR")
 # Prefer using UNIX domain sockets but if the `DEPUTY_IPC_DIR` is set assume the file
 # system is read-only and use a named pipe instead.
 function _graceful_terminator_socket_path(pid::Int32)
-    name = "graceful-terminator.$pid"
-    return haskey(ENV, "DEPUTY_IPC_DIR") ? joinpath(_deputy_ipc_dir(), name) : name
+    return joinpath(_deputy_ipc_dir(), "graceful-terminator.$pid")
```

Comment:
This now makes us always use named pipes. Not specifying a path should use a UNIX domain socket which don't necessary require. I'll want to run a couple of checks first.

Reply:
what's the issue with a named pipe vs. domain socket?

Reply:
https://stackoverflow.com/questions/35424970/unix-socket-permissions-linux

> In the Linux implementation, sockets which are visible in the filesystem honor the per‐ missions of the directory they are in. Their owner, group and their permissions can be changed. Creation of a new socket will fail if the process does not have write and search (execute) permission on the directory the socket is created in.

Reply:
I thought if you specified just a name in `listen` that a UNIX-domain socket would be created in `/run` automatically. It turns out that switch between UNIX-domain sockets and named pipes is a libuv thing: https://docs.libuv.org/en/v1.x/guide/processes.html#new-stdio-pipes.

I think this change is good to be consistent on where we write these files. It also seems that we should be [using `/run` more for both PID files and UNIX-domain sockets](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s15.html) so I opened #13 to handle these changes.